dotnet/src/Functions/Functions.OpenApi/Model/RestApiOperation.cs

```diff
@@ -119,24 +120,30 @@ public Uri BuildOperationUrl(IDictionary<string, string> arguments, Uri? serverU
     {
         var headers = new Dictionary<string, string>();
 
-        var headersMetadata = this.Parameters.Where(p => p.Location == RestApiOperationParameterLocation.Header);
+        var parameters = this.Parameters.Where(p => p.Location == RestApiOperationParameterLocation.Header);
 
-        foreach (var headerMetadata in headersMetadata)
+        foreach (var parameter in parameters)
         {
-            var headerName = headerMetadata.Name;
-
-            // Try to resolve header value in arguments.
-            if (arguments.TryGetValue(headerName, out string? value) && value is not null)
+            if (!arguments.TryGetValue(parameter.Name, out string? argument) || argument is null)
             {
-                headers.Add(headerName, value!);
+                //Throw an exception if the parameter is a required one but no value is provided.
```

Comment:
This two `ifs` can be combined

```diff
- if (!arguments.TryGetValue(parameter.Name, out string? argument) || argument is null)
- {
-     if (parameter.IsRequired)
+ if (parameter.IsRequired && (!arguments.TryGetValue(parameter.Name, out string? argument) || argument is null))
```

Reply:
They can't because we need to "skip" (the continue operator) non-required parameters, rather than just throwing exceptions.